tests/helpers/test_script.py

```diff
@@ -4084,3 +4085,145 @@ async def test_error_action(hass, caplog):
         },
         expected_script_execution="aborted",
     )
+
+
+async def test_continue_on_error(hass: HomeAssistant) -> None:
+    """Test if automation continue when a step fails."""
+    events = async_capture_events(hass, "test_event")
+
+    @callback
+    def broken_service(service: ServiceCall) -> None:
+        """Break this service with an error."""
+        raise HomeAssistantError("It is not working!")
+
+    hass.services.async_register("broken", "service", broken_service)
+
+    sequence = cv.SCRIPT_SCHEMA(
+        [
+            {"event": "test_event"},
+            {
+                "continue_on_error": True,
+                "service": "broken.service",
```

Comment:
Maybe also test that this `broken_service` would actually break the script without the `continue_on_error`?

Reply:
Done, I've split out the stop test that was hidden in here :)